### PR TITLE
Implement Stock Photos without WPMediaPicker

### PR DESCRIPTION
**Pull request by kean**
This PR reimplements Stocks Photos picker without the `WPMediaPicker` framework.

This is the final PR after which I can remove WPMediaPicker 😓 

## Changes

- Implement Stock photos picker using the new infra with `ExternalMediaPickerViewController`
- Change “Done” button to “Add”, including the Site Media picker
- Fix color of button “Show Selection”

## Testing

For every of the following flows:

- Add Image to Gutenberg
- Add Gallery to Gutenberg
- Add image to Aztec
- Add images to Site Media

Test the following:

- Verify that the new picker is displayed
- Verify that "Cancel" dismisses the screen and doesn't add any assets if the selection is made
- Verify that for multiple selection, the selection it oredered
- Verify that for multiple selection, a toolbar with a number of selected of items is displayed
- Verify that when you tap a toolbar button indicating the selection count the fullscreen gallery is displayed
- Verify that you can zoom in in the gallery
- Verify that when you tap "Done" the selected image are uploaded

## Regression Notes
1. Potential unintended areas of impact: Tenor (Aztec and Gutenberg post editor, Site Media screen)
2. What I did to test those areas of impact (or what existing automated tests I relied on): manual
3. What automated tests I added (or what prevented me from doing so): n/a

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

UI Changes testing checklist:
- [ ] Portrait and landscape orientations.
- [ ] Light and dark modes.
- [ ] Fonts: Larger, smaller and bold text.
- [ ] High contrast.
- [ ] VoiceOver.
- [ ] Languages with large words or with letters/accents not frequently used in English.
- [ ] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [ ] iPhone and iPad. 
- [ ] Multi-tasking: Split view and Slide over. (iPad)

### CODE REVIEW

File: WordPress/Classes/ViewRelated/Gutenberg/GutenbergViewController.swift [hunk 1]
```diff
@@ -16,15 +16,8 @@ class GutenbergViewController: UIViewController, PostEditor, FeaturedImageDelega
...
-        return GutenbergStockPhotos(gutenberg: gutenberg, mediaInserter: mediaInserterHelper)
-    }()
-    private lazy var filesAppMediaPicker: GutenbergFilesAppMediaSource = {
-        return GutenbergFilesAppMediaSource(gutenberg: gutenberg, mediaInserter: mediaInserterHelper)
-    }()
-    private lazy var tenorMediaPicker: GutenbergTenorMediaPicker = {
-        return GutenbergTenorMediaPicker(gutenberg: gutenberg, mediaInserter: mediaInserterHelper)
-    }()
+    private lazy var filesAppMediaPicker = GutenbergFilesAppMediaSource(gutenberg: gutenberg, mediaInserter: mediaInserterHelper)
+    private lazy var externalMediaPicker = GutenbergExternalMeidaPicker(gutenberg: gutenberg, mediaInserter: mediaInserterHelper)
```
Comment (momo-ozawa) [inline WordPress/Classes/ViewRelated/Gutenberg/GutenbergViewController.swift]:
```suggestion
    private lazy var externalMediaPicker = GutenbergExternalMediaPicker(gutenberg: gutenberg, mediaInserter: mediaInserterHelper)
```

File: WordPress/Classes/ViewRelated/Gutenberg/Utils/GutenbergExternalMeidaPicker.swift [hunk 2]
```diff
@@ -11,16 +11,24 @@ class GutenbergTenorMediaPicker {
...
-    func presentPicker(origin: UIViewController, post: AbstractPost, multipleSelection: Bool, callback: @escaping MediaPickerDidPickMediaCallback) {
+    func presentTenorPicker(origin: UIViewController, post: AbstractPost, multipleSelection: Bool, callback: @escaping MediaPickerDidPickMediaCallback) {
         mediaPickerCallback = callback
         self.multipleSelection = multipleSelection
 
         MediaPickerMenu(viewController: origin, isMultipleSelectionEnabled: multipleSelection)
             .showFreeGIFPicker(blog: post.blog, delegate: self)
     }
+
+    func presentStockPicker(origin: UIViewController, post: AbstractPost, multipleSelection: Bool, callback: @escaping MediaPickerDidPickMediaCallback) {
```
Comment (momo-ozawa) [inline WordPress/Classes/ViewRelated/Gutenberg/Utils/GutenbergExternalMeidaPicker.swift]: Wdyt of calling this presentStockPhotoPicker?

**Review by momo-ozawa — APPROVED** [review]

File: WordPress/Classes/ViewRelated/Gutenberg/Utils/GutenbergExternalMeidaPicker.swift (hunk 2 above)
Comment (kean) [reply WordPress/Classes/ViewRelated/Gutenberg/Utils/GutenbergExternalMeidaPicker.swift]: Updated, thanks!

File: WordPress/Classes/ViewRelated/Gutenberg/GutenbergViewController.swift (hunk 1 above)
Comment (kean) [reply WordPress/Classes/ViewRelated/Gutenberg/GutenbergViewController.swift]: Fixed

File: WordPress/Classes/ViewRelated/Media/StockPhotos/StockPhotosDataSource.swift [hunk 3]
```diff
@@ -17,25 +15,13 @@ final class StockPhotosDataSource: NSObject, WPMediaCollectionDataSource {
...
-        return 1
-    }
-
-    func search(for searchText: String?) {
-        searchQuery = searchText ?? ""
+    func search(for searchText: String) {
+        searchQuery = searchText
 
-        guard searchText?.isEmpty == false else {
+        guard searchText.count > 1 else {
```
Comment (oguzparlak) [inline WordPress/Classes/ViewRelated/Media/StockPhotos/StockPhotosDataSource.swift:24]: prefer ``isEmpty`` over ``count`` since its complexity is O(1).

**Pull request merged**